src/main/kotlin/nl/avisi/structurizr/site/generatr/ServeCommand.kt

```diff
@@ -111,7 +132,9 @@ class ServeCommand : Subcommand("serve", "Start a development server") {
         while (true) {
             val watchKey = watchService.take()
             val parentPath = watchKey.watchable() as Path
-            val events = watchKey.pollEvents()
+
+            Thread.sleep(100) // Throttle the file watcher to give editors time for cleaning up temporary files
```

Comment:
This might be a Linux / file system dependent thing.

Reply:
Did you have a particular reason for putting the `Thread.sleep` here, instead of at the end of the `while` loop?

Reply:
Yes, this was intentionally. 
From what I figured, `watchKey.pollEvents` gives you all events that have happened until now. So by putting the sleep here, we collect/drain all events between the first event that triggered `watchService.take()` and now (thus all events in the 100ms after the first event). Those 100ms where sufficient om my machine to let IntelliJ finish its file save operation (which contained multiple save actions) to not have any outstanding events left that would again trigger an update site.